### PR TITLE
Initialize UserStream

### PR DESCRIPTION
**Pull request by ericboucher**
I am trying to add new streams with `users` as the parent instead of `repositories`, but is proving to be a lot more complex than I anticipated.

So I am opening this draft PR to discuss the best strategy.

I started by splitting the streams into `repository_streams` and `user_streams`. But then I am not sure how to best differentiate them. Should it be a completely different tap?

The tests are definitely heavily skewed towards `repositories` at the moment. Would love your input @aaronsteers and @laurentS

### CODE REVIEW

File: tap_github/tap.py [hunk 1]
```diff
@@ -37,24 +41,39 @@ class TapGitHub(Tap):
...
         ),
         th.Property("repositories", th.ArrayType(th.StringType)),
+        th.Property("user_usernames", th.ArrayType(th.StringType)),
+        th.Property("user_ids", th.ArrayType(th.StringType)),
         th.Property("start_date", th.DateTimeType),
         th.Property("stream_maps", th.ObjectType()),
         th.Property("stream_map_config", th.ObjectType()),
     ).to_dict()
 
     def discover_streams(self) -> List[Stream]:
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tap_github/tap.py:51]: @ericboucher it's merged

**Comment by ericboucher** [comment]
@aaronsteers @edgarrmondragon ready for final review :)

File: tap_github/tap.py [hunk 2]
```diff
@@ -37,27 +41,42 @@ class TapGitHub(Tap):
...
+        VALID_QUERIES = VALID_REPO_QUERIES.union(VALID_USER_QUERIES)
+
+        if len(VALID_QUERIES.intersection(self.config)) != 1:
+            raise ValueError(
+                "This tap requires one and only one of the following path options: "
+                f"{VALID_QUERIES}."
+            )
+        is_user_query = len(VALID_USER_QUERIES.intersection(self.config)) > 0
+        if is_user_query:
+            return [UserStream(tap=self)]
```
Comment (edgarrmondragon) [inline tap_github/tap.py:66]: Should these operation "modes" be documented in the project's readme?

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

**Review by edgarrmondragon — APPROVED** [review]
lgtm!

**Comment by ericboucher** [comment]
@edgarrmondragon ready to go :)

**Pull request merged**